docs/posts/ibis-bench/index.qmd

```diff
@@ -0,0 +1,1313 @@
+
+Queries 16, 21, and 22 fail for the DataFusion backend via Ibis dataframe code,
+and query 16 fails through SQL. Note that [all TPC-H SQL queries successfully
+run through DataFusion
+directly](https://github.com/apache/datafusion-benchmarks) -- Ibis generates SQL
+that [hits a bug with DataFusion that has already been
+fixed](https://github.com/apache/datafusion/issues/10830).  We expect these
+queries to work in the next iteration of this benchmark coming soon.
+
+#### Failing Polars queries
+
+Queries 11, 13-17, and 20-22 fail for the Polars backend via Ibis dataframe
+code. These all fail due to lack of scalar subquery support in the backend. I've
+[opened an issue](https://github.com/ibis-project/ibis/issues/9422) for tracking
+and discussion.
+
+:::{.callout-tip title="Interested in contributing?"}
+Interesting in contributing to Ibis? Increasing coverage of operations for a
+backend is a great place to start!
+:::
+
+### How queries are written
+
+See [the source
+code](https://github.com/lostmygithubaccount/ibis-bench/tree/v2.0.0/src/ibis_bench/queries)
+for the exact queries used in this iteration of the benchmark. Polars recently
+updated their TPC-H queries, so the next iteration of this benchmark would use
+those.
+
+Queries were adapted from [Ibis TPC-H
+queries](https://github.com/ibis-project/ibis/tree/main/ibis/backends/tests/tpch)
+and [Polars TPC-H queries](https://github.com/pola-rs/tpch). The first 10 Ibis
+dataframe queries were translated from the Polars dataframe queries, while the
+rest were directly adapted from the Ibis repository. The SQL strings were
+adapted from the Ibis repository.
+
+### How queries are run
+
+See [the source
+code](https://github.com/lostmygithubaccount/ibis-bench/tree/v2.0.0) and
+[methodology](https://ibis-bench.streamlit.app/methodology) for more details. In
+short:
+
+- data is generated as a Parquet file per table
+    - standard DuckDB Parquet writer is used (snappy compression, ~100k row
+     group size, etc.)
+    - data is always downloaded onto a compute instance (no cloud storage reads)
+- decimal types are converted to floats after reading
+    - this is done to avoid several issues experienced
+    - in the next iteration of this benchmark, we'll use the `decimal` type
+- each query is run three times per configuration (system, scale factor, instance type)
+- we measure the time to write the results of the query to a Parquet file
+    - this includes reading the Parquet file(s) and executing the query
+
+### Biases
+
+My name is Cody and I'm a Senior Technical Product Manager at [Voltron
+Data](https://voltrondata.com). I am a contributor to the Ibis project and
+employed to work on it -- I'm biased in favor of Ibis and the composable data
+ecosystem.
+
+Ibis is [an independently governed open source
+project](https://github.com/ibis-project/governance) that **is not owned by
+Voltron Data**, though several steering committee members are employed by
```

Comment:
```suggestion
Voltron Data**, though several steering committee members are employed by
```

Linking to governance repo might be nice here, but not necessary.

Reply:
it is linked above

Reply:
I guess I can't read today!